<h1 style= "color:green"> <b>Activty 1:</b><span style= "color:blue"> PALMER PENGUIN DATA ANALYSIS PROJECT</span></h1>
<hr>

<p><b>Analytics: Dwight Francis M. Caña</b></p>


In [1]:
#libraries

In [52]:
import numpy as np
import pandas as pd
import warnings
import scipy.stats as onesome
warnings.filterwarnings('ignore')

In [3]:
berd = pd.read_csv('datasets\\penguins.csv')

In [4]:
berd

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<hr>
<h1 style="color:green">Descriptive Analytics</h1>

In [5]:
berd.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [6]:
berd.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<hr>
<H1>Filtering a Column/Data</H1>

In [7]:
berd['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [8]:
berd['species']=="Adelie"

0       True
1       True
2       True
3       True
4       True
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [9]:
Adelie_df = berd[berd['species']=="Adelie"]

In [10]:
Gentoo_df = berd[berd['species']=="Gentoo"]

In [11]:
Chinstrap_df = berd[berd['species']=="Chinstrap"]

In [12]:
Adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [13]:
Gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [14]:
Chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

<hr>


<h2>Averages</h2>

In [15]:
Adelie_df['body_mass_g'].mean()

np.float64(3700.662251655629)

In [16]:
Gentoo_df['body_mass_g'].mean()

np.float64(5076.016260162602)

In [17]:
Chinstrap_df['body_mass_g'].mean()

np.float64(3733.0882352941176)

In [18]:
berd[(berd['species']=="Adelie") & (berd['island']=="Dream")]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30   Adelie  Dream            39.5           16.7              178.0   
31   Adelie  Dream            37.2           18.1              178.0   
32   Adelie  Dream            39.5           17.8              188.0   
33   Adelie  Dream            40.9           18.9              184.0   
34   Adelie  Dream            36.4           17.0              195.0   
35   Adelie  Dream            39.2           21.1              196.0   
36   Adelie  Dream            38.8           20.0              190.0   
37   Adelie  Dream            42.2           18.5              180.0   
38   Adelie  Dream            37.6           19.3              181.0   
39   Adelie  Dream            39.8           19.1              184.0   
40   Adelie  Dream            36.5           18.0              182.0   
41   Adelie  Dream            40.8           18.4              195.0   
42   Adelie  Dream            36.0           18.5              186.0   
43   Adelie  Dream            44.1           19.7              196.0   
44   Adelie  Dream            37.0           16.9              185.0   
45   Adelie  Dream            39.6           18.8              190.0   
46   Adelie  Dream            41.1           19.0              182.0   
47   Adelie  Dream            37.5           18.9              179.0   
48   Adelie  Dream            36.0           17.9              190.0   
49   Adelie  Dream            42.3           21.2              191.0   
84   Adelie  Dream            37.3           17.8              191.0   
85   Adelie  Dream            41.3           20.3              194.0   
86   Adelie  Dream            36.3           19.5              190.0   
87   Adelie  Dream            36.9           18.6              189.0   
88   Adelie  Dream            38.3           19.2              189.0   
89   Adelie  Dream            38.9           18.8              190.0   
90   Adelie  Dream            35.7           18.0              202.0   
91   Adelie  Dream            41.1           18.1              205.0   
92   Adelie  Dream            34.0           17.1              185.0   
93   Adelie  Dream            39.6           18.1              186.0   
94   Adelie  Dream            36.2           17.3              187.0   
95   Adelie  Dream            40.8           18.9              208.0   
96   Adelie  Dream            38.1           18.6              190.0   
97   Adelie  Dream            40.3           18.5              196.0   
98   Adelie  Dream            33.1           16.1              178.0   
99   Adelie  Dream            43.2           18.5              192.0   
132  Adelie  Dream            36.8           18.5              193.0   
133  Adelie  Dream            37.5           18.5              199.0   
134  Adelie  Dream            38.1           17.6              187.0   
135  Adelie  Dream            41.1           17.5              190.0   
136  Adelie  Dream            35.6           17.5              191.0   
137  Adelie  Dream            40.2           20.1              200.0   
138  Adelie  Dream            37.0           16.5              185.0   
139  Adelie  Dream            39.7           17.9              193.0   
140  Adelie  Dream            40.2           17.1              193.0   
141  Adelie  Dream            40.6           17.2              187.0   
142  Adelie  Dream            32.1           15.5              188.0   
143  Adelie  Dream            40.7           17.0              190.0   
144  Adelie  Dream            37.3           16.8              192.0   
145  Adelie  Dream            39.0           18.7              185.0   
146  Adelie  Dream            39.2           18.6              190.0   
147  Adelie  Dream            36.6           18.4              184.0   
148  Adelie  Dream            36.0           17.8              195.0   
149  Adelie  Dream            37.8           18.1              193.0   
150  Adelie  Dream            36.0          

In [19]:
berd[(berd['species']=="Gentoo") & (berd['year'] <= 2008)]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
227  Gentoo  Biscoe            48.6           16.0              230.0   
228  Gentoo  Biscoe            47.5           14.2              209.0   
229  Gentoo  Biscoe            51.1           16.3              220.0   
230  Gentoo  Biscoe            45.2           13.8              215.0   
231  Gentoo  Biscoe            45.2           16.4              223.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
227       5800.0    male  2008  
228       4600.0  female  2008  
229       6000.0    male  2008  
230       4750.0  female  2008  
231       5950.0    male  2008  

[80 rows x 8 columns]

<hr>
<h1>Cleaning the <i>Data Frame~</i></h1>

In [20]:
berd.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<hr style= 'color:'>
<h1>What to do with <b>Null</b> Values</h1>

In [21]:
berd['bill_length_mm']=berd['bill_length_mm'].fillna(berd['bill_length_mm'].mean())
berd['bill_depth_mm']=berd['bill_depth_mm'].fillna(berd['bill_depth_mm'].mean())
berd['flipper_length_mm']=berd['flipper_length_mm'].fillna(berd['flipper_length_mm'].mean())
berd['body_mass_g']=berd['body_mass_g'].fillna(berd['body_mass_g'].mean())
berd['sex']=berd['sex'].fillna(berd['sex'].mode()[0])

In [22]:
berd.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<hr>
<h1>Data Types</h1>

In [23]:
berd.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [24]:
berd.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      344.000000     344.000000         344.000000   344.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.443643       1.969027          14.020657   799.613058   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.275000      15.600000         190.000000  3550.000000   
50%         44.250000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<hr>
<h1>Changing Data Types</h1>

In [25]:
berd['species'] = berd['species'].astype('category')
berd['island'] = berd['island'].astype('category')
berd['sex'] = berd['sex'].astype('category')
berd['year'] = berd['year'].astype('category')

In [26]:
berd.describe()


bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      344.000000     344.000000         344.000000   344.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.443643       1.969027          14.020657   799.613058
min         32.100000      13.100000         172.000000  2700.000000
25%         39.275000      15.600000         190.000000  3550.000000
50%         44.250000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

In [27]:
berd.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<hr>
<h1><span style= "color:red">Activity 1:</span> 7 questions</h1>

<h2>1. which species of penguin is the heaviest?  </h2>

In [28]:
Adelie_df['body_mass_g'].mean()

np.float64(3700.662251655629)

In [29]:
Gentoo_df['body_mass_g'].mean()

np.float64(5076.016260162602)

In [30]:
Chinstrap_df['body_mass_g'].mean()

np.float64(3733.0882352941176)

<h4>Insight No. 1</h4>
Genton is the heaviest
<hr>

<h2>2. which sex of penguin has the longer flippers?</h2>

In [31]:
berd[berd['sex']=="male"]['flipper_length_mm'].mean()

np.float64(204.18899670031692)

In [32]:
berd[berd['sex']=="female"]['flipper_length_mm'].mean()

np.float64(197.36363636363637)

<h4>Insight No.2</h4>
males has the longer flippers
<hr>

<h2>3. which island has the heaviest penguins</h2>

In [33]:
berd['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

In [34]:
berd[berd['island']=="Biscoe"]['body_mass_g'].mean()

np.float64(4712.95687134503)

In [35]:
berd[berd['island']=="Dream"]['body_mass_g'].mean()

np.float64(3712.9032258064517)

In [36]:
berd[berd['island']=="Torgersen"]['body_mass_g'].mean()

np.float64(3715.8991228070176)

<h4>Insight No.3</h4>
The island biscoe has the heaviest penguins
<hr>

<h2>4. which sex of penguin is heaviest?</h2>

In [37]:
berd[berd['sex']=="male"]['body_mass_g'].mean()

np.float64(4514.684406547094)

In [38]:
berd[berd['sex']=="female"]['body_mass_g'].mean()

np.float64(3862.2727272727275)

<h4>Insight No.4</h4>
Males are heavier than the females
<hr>

<h2>5. which year has the penguin population been the biggest</h2>


In [39]:
berd['year'].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h4>Insight No.5</h4>
the penguins populations is the highets during 2009 comparatively to 2008 and 2007 so the population is growing
<hr>

<h2>6. what are the lightest and heaviest penguin in both genders?</h2>

In [40]:
berd[berd['sex']=="male"]['body_mass_g'].min()

np.float64(2975.0)

In [41]:
berd[berd['sex']=="male"]['body_mass_g'].max()

np.float64(6300.0)

In [42]:
berd[berd['sex']=="female"]['body_mass_g'].max()

np.float64(5200.0)

In [43]:
berd[berd['sex']=="female"]['body_mass_g'].min()

np.float64(2700.0)

<h4>Insight No.6</h4>
the heaviest male & female penguin is 6.3kg and 5.2kg
the lightest male & female penguin is 2.975g and 2.7kg
<hr>

<h2>7. Which island has the most same species of penguins
</h2>

In [44]:
seven= berd.groupby(['species','island']).size()

In [45]:
seven

species    island   
Adelie     Biscoe        44
           Dream         56
           Torgersen     52
Chinstrap  Biscoe         0
           Dream         68
           Torgersen      0
Gentoo     Biscoe       124
           Dream          0
           Torgersen      0
dtype: int64

<h4>Insight No.7</h4>
both Chinstrap and Gentoo only occupy one island respectively. Dream for chinstrap, Biscoe for Gentoo with gentoo having 124 population in biscoe and Adelie with 44.
<p></p>
Biscoe houses the most amount of penguins overall aswell to housing the most of the same species(gentoo)
<hr>

<h2>8. Do penguin penguin length related to bill depth
</h2>

In [78]:
beaks=berd['bill_length_mm'].corr(berd['bill_depth_mm'])

In [79]:
beaks

np.float64(-0.23505287035553282)

<h4>Insight No.8</h4>
seems like the length of the bill is unrelated to he bill depth
<hr>

<h2>9. Which penguin species has the longest flippers

In [80]:
baf=berd[berd['species']=="Adelie"]['flipper_length_mm']
bgf=berd[berd['species']=="Gentoo"]['flipper_length_mm']
bcf=berd[berd['species']=="Chinstrap"]['flipper_length_mm']

In [81]:
baf.mean()

np.float64(190.02575792551556)

In [82]:
bgf.mean()

np.float64(217.05576777966422)

In [83]:
bcf.mean()

np.float64(195.8235294117647)

<h4>Insight No.9</h4>
gento being the heaviest bird has also the longest flippers
<hr>

<h2>10. Does being fat make you bill have more depth?


In [101]:
fat=berd['body_mass_g'].corr(berd['bill_depth_mm'])

In [102]:
fat

np.float64(-0.47191562118606645)

<h4>Insight No.10</h4>
being heavy doesnt effect the depth of penguins bill
<hr>

<h2>11. Does being fat make your bill longer?

In [103]:
long=berd['body_mass_g'].corr(berd['bill_length_mm'])

In [104]:
long

np.float64(0.5951098244376305)

<h4>Insight No.11</h4>
The weight of the penguin is connected to the lenght of the bill unlike the depth
<hr>

<h2>12. Does being heavy connected to flipper lenght

In [108]:
wing=berd['body_mass_g'].corr(berd['flipper_length_mm'])

In [109]:
wing

np.float64(0.8712017673060114)

<h4>Insight No.12</h4>
The weight of the penguin is connected to the lenght of the flippers
<hr>